### PR TITLE
Winrm hold mutex for install

### PR DESCRIPTION
**Pull request by duncangrant**
This PR prevents processes concurrently trying to install items on the same winrm location.  It is as similar as possible to how we handle ssh.

### CODE REVIEW

File: software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessWinRmDriver.java [hunk 1]
```diff
@@ -154,6 +171,38 @@ public void runPostLaunchCommand() {
...
+
+    @Override
+    public void copyInstallResources() {
+        final WithMutexes mutexSupport = getLocation().mutexes();
+        String mutexId = "installation lock at host";
+        mutexSupport.acquireMutex(mutexId, "installing " + elvis(entity, this));
+        try {
+            super.copyInstallResources();
+        } catch (Exception e) {
+//            log.warn("Error copying install resources", e);
```
Comment (aledsage) [inline software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessWinRmDriver.java]: Remove commented out code?

File: software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessWinRmDriver.java [hunk 2]
```diff
@@ -154,6 +171,38 @@ public void runPostLaunchCommand() {
...
+            throw Exceptions.propagate(e);
+        } finally {
+            mutexSupport.releaseMutex(mutexId);
+        }
+    }
+
+    @Override
+    public void copyCustomizeResources() {
```
Comment (aledsage) [inline software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessWinRmDriver.java]: What about `copyPreInstallResources`? Should we override that? No strong feelings whether we override `copyCustomizeResources` or not (we don't for ssh I think).

File: software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessWinRmDriver.java [hunk 3]
```diff
@@ -154,6 +171,38 @@ public void runPostLaunchCommand() {
...
+        final WithMutexes mutexSupport = getLocation().mutexes();
+        final String description = "customizing " + elvis(entity, this);
+        String mutexId = "installation lock at host";
+        mutexSupport.acquireMutex(mutexId, description);
+        try {
+            super.copyCustomizeResources();
+        } catch (Exception e) {
+//            log.warn("Error copying customize resources", e);
```
Comment (aledsage) [inline software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessWinRmDriver.java]: Again delete commented out code?

**Comment by duncangrant** [comment]
@aledsage I think I've addressed all comments and mirrored the changes for ssh drivers.

**Comment by aledsage** [comment]
LGTM; merging.

**Pull request merged**